python/ray/train/_internal/state/state_manager.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import os
+from typing import Dict
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.state.schema import (
+    TrainDatasetInfo,
+    TrainRunInfo,
+    TrainWorkerInfo,
+)
+from ray.train._internal.state.state_actor import (
+    TRAIN_STATE_ACTOR_NAME,
+    TRAIN_STATE_ACTOR_NAMESPACE,
+)
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+class TrainRunStateManager:
+    """A class that aggregates and reports train run info to TrainStateActor.
+
+    This manager class is created on the train controller layer for each run.
+    """
+
+    def __init__(self, worker_group: WorkerGroup) -> None:
+        self.state_actor = ray.get_actor(
+            name=TRAIN_STATE_ACTOR_NAME, namespace=TRAIN_STATE_ACTOR_NAMESPACE
+        )
+        self.worker_group = worker_group
```

Comment:
Can we just pass in worker group into the `register_train_run`? No need to keep it around as a property.

Reply:
Done.